example/macos/Runner.xcodeproj/project.pbxproj

```diff
@@ -366,7 +367,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "\"$FDE_ROOT\"/tools/build_flutter_assets \"$PROJECT_DIR\"/..\n";
+			shellScript = "\"$FLUTTER_ROOT\"/packages/flutter_tools/bin/macos_build_flutter_assets.sh";
```

Comment:
This creates a new requirement that `flutter build` must be run before the Xcode project will work; that's not an issue for a long-term `flutter create`d version of this, but would be potentially breaking for this project. I think we should check in a Generated.xcconfig with this change that defaults to `../../../flutter` to make it work in a clean checkout that uses the suggested layout. (Not perfect, since it won't handle .flutter_location_config, but better than nothing.)

Reply:
Putting it together this way it doesn't quite work when i run from Xcode - possibly due to the lack of the OBJROOT, SYMROOT, and derivedData overrides?  I might have my setup configured wrong.

Reply:
I'll take a look tomorrow and see what's broken. The build output will go somewhere else if you build from Xcode, but I wouldn't expect that to break anything (it's what happens right now with build.sh vs Xcode UI).

Reply:
I realized while doing the Windows version that we shouldn't check in a copy of the generated settings, since it needs to be in the .gitignore, so I think we'll just live with needing to use 'fluter build' or 'flutter run' at least once. I can add a note about that to the docs.